db/index.js

```diff
@@ -36,29 +37,246 @@ async function getAllProducts() {
 
 
 async function createProduct(product) {
-
   const { name, description, price, imageURL, inStock, category } = product
-  
   try {
-   
     const { rows: [ newProduct ] } = await client.query(`
       INSERT INTO products(name, description, price, imageURL, "inStock", category) 
       VALUES($1, $2, $3, $4, $5, $6) 
       RETURNING *;
     `, [name, description, price, imageURL, inStock, category]);
 
-       return newProduct;
+    return newProduct;
+  } catch (error) {
+    throw error;
+  }
+};
+
+
+async function createUser({firstName, lastName, email, imageurl, username, password, isAdmin}) {
+  const hashedPassword = await bcrypt.hash(password, SALT_COUNT);
```

Comment:
Create a "safe list" of only the columns that are required, and if any of those isn't defined, then throw an error...
This is not a perfect system... but something like this.
```js
const required = [firstName, lastName, email, username, password];
const notFound = required.filter(requiredItem => !params[requiredItem]);
for(let missing of notFound) {
  // add to some error list
}
```